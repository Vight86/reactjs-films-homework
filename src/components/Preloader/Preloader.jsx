import React from 'react';
import styles from './Preloader.scss';

const Preloader = () => (
  <div className={styles.preloader}>
    <div className={styles.preloader__circles}>
      <span className={styles.preloader__circle} />
      <span className={styles.preloader__circle} />
      <span className={styles.preloader__circle} />
    </div>
    <p className={styles.preloader__text}>Loading</p>
  </div>
);

export default Preloader;
