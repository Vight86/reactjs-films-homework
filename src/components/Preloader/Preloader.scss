@import '../../utils/variables';

.preloader {
  width: 12em;
  margin: 1em auto;
  text-align: center;

  &__circles {
    position: relative;
    width: 12em;
    height: 3em;
  }

  &__circle {
    position: absolute;
    left: 1.1em;
    display: inline-block;
    width: 1.8em;
    height: 1.8em;
    margin-top: 1em;
    background-color: $color-secondary;
    border-radius: 50%;
    transform: translateX(0) scale(0);

    &:nth-child(1) {
      animation: loading 2.5s ease-in-out infinite;
    }

    &:nth-child(2) {
      animation: loading 2s ease-in-out 0.5s infinite;
    }

    &:nth-child(3) {
      animation: loading 2s ease-in-out 0.7s infinite;
    }
  }

  &__text {
    margin: 1em 0 2em;
    color: $text-tertiary;
    font-size: 2.2em;
    font-weight: 500;
    letter-spacing: 0.04em;
    opacity: 0.35;
    text-transform: uppercase;
  }
}

@keyframes loading {
  50% {
    transform: translateX(4em) scale(1);
  }

  100% {
    transform: translateX(8em) scale(0);
  }
}
