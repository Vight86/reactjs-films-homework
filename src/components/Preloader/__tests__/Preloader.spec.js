import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Preloader from '../index';

const setUp = () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Preloader />);
  return renderer.getRenderOutput();
};

describe('render Preloader component', () => {
  it('render component correctly', () => {
    const result = setUp();
    expect(result).toMatchSnapshot();
  });
});
