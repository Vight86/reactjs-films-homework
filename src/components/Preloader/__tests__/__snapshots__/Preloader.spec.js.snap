// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Preloader component render component correctly 1`] = `
<div
  className="preloader"
>
  <div
    className="preloader__circles"
  >
    <span
      className="preloader__circle"
    />
    <span
      className="preloader__circle"
    />
    <span
      className="preloader__circle"
    />
  </div>
  <p
    className="preloader__text"
  >
    Loading
  </p>
</div>
`;
